Hardware::CPU.type: check uname -m

xu-cheng opened the pull request:
Determining CPU type by reading `/proc/cpuinfo` is less reliable than
checking the value of `uname -m`. Particularly, `/proc/cpuinfo` often
return the CPU info for the host CPU when running Linuxbrew inside a VM.
Therefore, if the host and VM share different CPU architectures,
`/proc/cpuinfo` will return incorrect result.

To fix the problem, checking `uname -m` first, otherwise fallback to
`/proc/cpuinfo`.

Fixes #336.

sjackman commented on Library/Homebrew/extend/os/linux/hardware/cpu.rb:
```diff
@@ -20,12 +20,19 @@ def cpuinfo
+          elsif cpuinfo =~ /Intel|AMD/
+            :intel
+          elsif cpuinfo =~ /ARM|Marvell/
+            :arm
```
I'm not a fan of the redundancy. I did imply that's what I wanted. Sorry for that.

Let's delete lines 29–32 until we find a situation where they're necessary. What does `uname -m` return on a 32-bit x86 system? As I recall it matches `/86$/`, like `i386`, `i586` et c.
I oddly can't find the original issue for the `Marvel` ARM system. I'm fine ignoring that for now.
Let's use
```ruby
case Utils.popen_read("uname", "-m").chomp
```

xu-cheng replied:
Updated

sjackman reviewed (changes requested):
Minor changes requested. Works for me, in principle. Thanks again, Xu.

xu-cheng committed:
Hardware::CPU.type: check uname -m

Determining CPU type by reading `/proc/cpuinfo` is less reliable than
checking the value of `uname -m`. Particularly, `/proc/cpuinfo` often
return the CPU info for the host CPU when running Linuxbrew inside a VM.
Therefore, if the host and VM share different CPU architectures,
`/proc/cpuinfo` will return incorrect result.

To fix the problem, using `uname -m` for CPU architecture detection.

Fixes #336.

sjackman commented:
Thanks again for your contribution to Linuxbrew, Xu!
